mktags.bash

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+# Icinga 2 Docker image | (c) 2023 Icinga GmbH | GPLv2+
+set -exo pipefail
+
+if [[ "$1" =~ ^v((([0-9]+).([0-9]+)).[0-9]+)$ ]]; then
+  BUILDX=(docker buildx build --platform "$(cat "$(realpath "$(dirname "$0")")/platforms.txt")" --push)
+  cd "$(mktemp -d)"
+
+  cat <<<"FROM icinga/icinga2:${BASH_REMATCH[1]}" >Dockerfile
+
+  "${BUILDX[@]}" -t "icinga/icinga2:${BASH_REMATCH[2]}" .
+
+  docker pull "icinga/icinga2:${BASH_REMATCH[2]}"
```

Comment:
What's the purpose of this pull here?

Reply:
Oh, so you're using that to check that using pull to test whether a tag exists is not completely broken?

I'd suggest using

```sh
curl --head -s -o /dev/null -w "%{http_code}" https://hub.docker.com/v2/namespaces/icinga/repositories/icinga2/tags/$tag
```

and on

- 404: create the new tag
- 200: don't create the tag
- everything else: error out

[API docs](https://docs.docker.com/docker-hub/api/latest/#tag/repositories/paths/~1v2~1namespaces~1%7Bnamespace%7D~1repositories~1%7Brepository%7D~1tags~1%7Btag%7D/head)

Reply:
I prefer my version or even -if you prefer- no check for this at all.

Both do more with less.

Reply:
How does `docker pull` do more? The way you're using it, your just checking if it worked or not. But you can't distinguish why it didn't work, so if Docker Hub fails a single request with a 5xx, `docker pull` wouldn't detect this and publish a wrong tag. The `curl` API call on the other hand can distinguish these situations.